pkg/util/httpreq/tags.go

```diff
@@ -33,3 +35,22 @@ func ExtractQueryTagsMiddleware() middleware.Interface {
 		})
 	})
 }
+
+func ExtractQueryMetricsMiddleware() middleware.Interface {
+	return middleware.Func(func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			ctx := req.Context()
+
+			enqueueTimeHeaders := req.Header[string(QueryEnqueueTimeHTTPHeader)]
```

Comment:
followup to the prior nit:
```suggestion
			enqueueTimeHeaders := req.Header.Get(string(QueryEnqueueTimeHTTPHeader))
```

Reply:
Addressed in https://github.com/grafana/loki/pull/4949/commits/e54455730662767e894e453bfc5b2e75647e5dbe